tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1482 @@
+    implicit wp: WskProps): RestResult = {
+    val entPath = Path(s"$basePath/namespaces")
+    val resp = getWhiskEntity(entPath).futureValue
+    val result = if (resp == None) new RestResult(NotFound, null) else new RestResult(resp.status, getRespData(resp))
+    if (expectedExitCode != DONTCARE_EXIT)
+      if (result.statusCode.intValue != expectedExitCode)
+        result.exitCode shouldBe expectedExitCode
+    result
+  }
+
+  /**
+   * Gets entities in namespace.
+   *
+   * @param namespace (optional) if specified must be  fully qualified namespace
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def get(namespace: Option[String] = None,
+                   expectedExitCode: Int = OK.intValue,
+                   nameSort: Option[Boolean] = None)(implicit wp: WskProps): RestResult = {
+    val (ns, _) = namespace map { this.getNamespaceActionName(_) } getOrElse (wp.namespace, "")
+    val entPath = Path(s"$basePath/namespaces/$ns/")
+    val resp = getWhiskEntity(entPath).futureValue
+    val r = new RestResult(resp.status, getRespData(resp))
+    if (expectedExitCode != DONTCARE_EXIT) {
+      if (r.statusCode.intValue != expectedExitCode)
+        r.statusCode.intValue shouldBe expectedExitCode
+    }
+    r
+  }
+
+  /**
+   * Looks up namespace for whisk props.
+   *
+   * @param wskprops instance of WskProps with an auth key to lookup
+   * @return namespace as string
+   */
+  override def whois()(implicit wskprops: WskProps): String = {
+    val ns = list().getBodyListString
+    if (ns.size > 0) ns(0).toString() else ""
+  }
+}
+
+class WskRestPackage
+    extends RunWskRestCmd
+    with ListOrGetFromCollectionRest
+    with DeleteFromCollectionRest
+    with BasePackage {
+
+  override protected val noun = "packages"
+
+  /**
+   * Creates package. Parameters mirror those available in the REST.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def create(name: String,
+                      parameters: Map[String, JsValue] = Map(),
+                      annotations: Map[String, JsValue] = Map(),
+                      parameterFile: Option[String] = None,
+                      annotationFile: Option[String] = None,
+                      shared: Option[Boolean] = None,
+                      update: Boolean = false,
+                      expectedExitCode: Int = OK.intValue)(implicit wp: WskProps): RestResult = {
+    val path = Path(s"$basePath/namespaces/${wp.namespace}/$noun/$name")
+    var bodyContent = JsObject("namespace" -> s"${wp.namespace}".toJson, "name" -> name.toJson)
+
+    val (params, annos) = this.getParamsAnnos(parameters, annotations, parameterFile, annotationFile)
```

Comment:
Hmm, I don't understand why the parameter and annotations handling differs here base on action update/create.

Reply:
We pick up either parameters or parameterFile as the parameters to create the action, and pick up either annotations or annotationFile as the annotations to create the action. 
When it comes to update the action, we only care about the parameters and annotations, not parameterFile or annotationFile any more. That is why we see the differences.